packages/web-components/src/components/ic-toast/ic-toast.tsx

```diff
@@ -0,0 +1,318 @@
+  /**
+   * If toast dismissMode is set to `automatic`, use this prop to define the time before the toast dismisses (in MILLISECONDS)
+   * (NOTE: Has a minimum value of `5000ms`)
+   */
+  @Prop() autoDismissTimeout? = 5000;
+  /**
+   * Provides a custom alt-text to be announced to screen readers, if slotting a custom neutral icon
+   */
+  @Prop() neutralIconAltText?: string;
+  /**
+   * Is emitted when the user dismisses the toast
+   */
+  @Event() toastDismiss: EventEmitter<void>;
+
+  @State() visible = false;
+  @State() timerProgress = 100;
+
+  dismissTimeout: number;
+  timerRefreshInterval: number;
+  isManual: boolean;
+  interactiveElements: ActionAreaElementTypes[] = [];
+
+  /**
+   * Used to display the individual toast
+   * @returns The element that previously had focus before the toast appeared
+   */
+  @Method()
+  async setVisible() {
+    if (!this.visible) this.visible = true;
+    if (!this.isManual) {
+      this.dismissTimeout = window.setTimeout(
+        this.dismissAction,
+        this.autoDismissTimeout
+      );
+      this.timerRefreshInterval = window.setInterval(
+        this.handleProgressChange,
+        AUTO_DISMISS_TIMER_REFRESH_RATE_MS
+      );
+      return null;
+    } else {
+      return document.activeElement as HTMLElement;
+    }
+  }
+
+  @Listen("toastDismiss", { capture: true })
+  handleDismiss(): void {
+    this.visible = false;
+    clearInterval(this.timerRefreshInterval);
+    this.timerProgress = 100;
+  }
+
+  @Listen("keydown", { target: "document" })
+  handleKeyboard(ev: KeyboardEvent): void {
+    if (this.isManual && this.visible) {
+      switch (ev.key) {
+        case "Tab":
+          ev.preventDefault();
+          this.focusInteractiveEl(this.findNextInteractiveElement(ev.shiftKey));
+          break;
+        case "Escape":
+          !ev.repeat && this.dismissAction();
+          ev.stopImmediatePropagation();
+          break;
+      }
+    }
+  }
+
+  @Listen("mouseenter")
+  @Listen("mouseleave")
+  handleTimer(ev: MouseEvent): void {
+    if (!this.isManual) {
+      switch (ev.type) {
+        case "mouseenter":
+          window.clearTimeout(this.dismissTimeout);
+          window.clearInterval(this.timerRefreshInterval);
+          this.timerProgress = 100;
+          break;
+        case "mouseleave":
+          this.dismissTimeout = window.setTimeout(
+            this.dismissAction,
+            this.autoDismissTimeout
+          );
+          this.timerRefreshInterval = window.setInterval(
+            this.handleProgressChange,
+            AUTO_DISMISS_TIMER_REFRESH_RATE_MS
+          );
+          break;
+      }
+    }
+  }
+
+  private dismissAction = (): void => {
+    this.toastDismiss.emit();
+  };
+
+  private handleProgressChange = () => {
+    this.timerProgress -=
+      (AUTO_DISMISS_TIMER_REFRESH_RATE_MS / this.autoDismissTimeout) * 100;
+  };
+
+  private handleLongText(heading: boolean, message?: boolean): void {
```

Comment:
Can the arguments be more descriptive as they do not contain the heading or message.